codec/json/codec.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2021 - The Event Horizon authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+
+	eh "github.com/looplab/eventhorizon"
+)
+
+// EventCodec is a codec for marshaling and unmarshaling events
+// to and from bytes in BSON format.
```

Comment:
This file needs a global string replace of BSON -> JSON

Reply:
Oh, snap! 😕